src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -103,18 +107,13 @@ internal class ConnectionManager : IConnectionManager {
             return isRemoved;
         }
 
-        public async Task ConnectAsync(string name, string path, string rCommandLineArguments) {
-            var connection = GetOrCreateConnection(name, path, rCommandLineArguments);
-            await ConnectAsync(connection);
-        }
-
-        public async Task ConnectAsync(IConnection connection) {
+        public async Task ConnectAsync(IConnectionInfo connection) {
             var sessions = _sessionProvider.GetSessions().ToList();
             if (sessions.Any()) {
                 await Task.WhenAll(sessions.Select(s => s.StopHostAsync()));
             }
 
-            if (ActiveConnection != null && ActiveConnection.Id != connection.Id) {
+            if (ActiveConnection != null && (!ActiveConnection.Path.EqualsIgnoreCase(connection.Path)  || _sessionProvider.BrokerUri.IsLoopback)) {
```

Comment:
This fixes the issue when user had zero connections and no local R. Allows to switch away from the null/stub connector (`NullRHostConnector`). I am open to alternatives, I want to improve the class anyway so it throw more descriptive exception than just `RHostDisconnectedException`